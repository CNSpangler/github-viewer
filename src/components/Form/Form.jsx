import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ username, onUsernameChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="input" name="username" value={username} onChange={onUsernameChange}/>
    <button>Search</button>
  </form>
);

Form.propTypes = {
  username: PropTypes.string.isRequired,
  onUsernameChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;
