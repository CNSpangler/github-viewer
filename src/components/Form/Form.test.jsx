import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.jsx';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form 
      username="cnspangler"
      onUsernameChange={() => {}}
      onButtonClick={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
