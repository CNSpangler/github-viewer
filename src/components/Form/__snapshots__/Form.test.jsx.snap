// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="username"
    onChange={[Function]}
    type="input"
    value="cnspangler"
  />
  <button>
    Search
  </button>
</form>
`;
